php/php.blade/src/org/netbeans/modules/php/blade/editor/BladeBracesMatcher.java

```diff
@@ -0,0 +1,288 @@
+import org.netbeans.modules.php.blade.syntax.antlr4.v10.BladeAntlrUtils;
+import org.netbeans.spi.editor.bracesmatching.BracesMatcher;
+import org.netbeans.spi.editor.bracesmatching.BracesMatcherFactory;
+import org.netbeans.spi.editor.bracesmatching.MatcherContext;
+
+/**
+ * brace matcher - block directives : @if @endif .. - output echo statements {{
+ * }} {!! !!}
+ *
+ * @author bogdan
+ */
+public class BladeBracesMatcher implements BracesMatcher {
+
+    public enum BraceDirectionType {
+        END_TO_START, START_TO_END, CUSTOM_START_TO_END, CURLY_END_TO_START, CURLY_START_TO_END, STOP
+    }
+    private final MatcherContext context;
+    private Token originToken;
+    private BraceDirectionType currentDirection;
+
+    private BladeBracesMatcher(MatcherContext context) {
+        this.context = context;
+    }
+
+    @Override
+    public int[] findOrigin() throws InterruptedException, BadLocationException {
+        int[] result = null;
+        originToken = null;
+        BaseDocument document = (BaseDocument) context.getDocument();
+        document.readLock();
+        try {
+            Token currentToken = BladeAntlrUtils.getToken(context.getDocument(), context.getSearchOffset());
+
+            if (currentToken == null) {
+                return result;
+            }
+
+            if (!shouldLookForBraceMatch(currentToken)) {
+                return result;
+            }
+
+            originToken = currentToken;
+            int start = currentToken.getStartIndex();
+            int end = currentToken.getStopIndex();
+
+            String tokenText = originToken.getText();
+            
+            if (!tokenText.startsWith("@")
+                    && !tokenText.startsWith("{")
+                    && !tokenText.endsWith("}")) {
+                return result;
+            }
+
+            currentDirection = findBraceDirectionType(tokenText, currentToken);
+
+            if (currentDirection.equals(BraceDirectionType.STOP)) {
+                return result;
+            }
+
+            result = new int[]{start, end + 1};
+        } finally {
+            document.readUnlock();
+        }
+        return result;
+    }
+
+    @Override
+    public int[] findMatches() throws InterruptedException, BadLocationException {
+        int[] result = null;
+        if (originToken == null) {
+            return result;
+        }
+        String tokenText = originToken.getText();
+
+        switch (currentDirection) {
+            case CURLY_START_TO_END:
+                return findCloseTag();
+            case CURLY_END_TO_START:
+                return findOpenTag();
+            case START_TO_END:
+                return findDirectiveEnd(tokenText);
+            case CUSTOM_START_TO_END:
+                return findCustomDirectiveEnd(tokenText);
+            case END_TO_START:
+                return findOriginForDirectiveEnd(tokenText);
+        }
```

Comment:
Please add `default` case to switch statements. (We are not sure whether it is intentional.)
We can add `assert` if needed. (We can notice when new variables are added.)
In this case, the following?
```java
default: 
    return null;
```

Reply:
From a general IDE perspective if moving to JDK17 - for enums, particularly internal to a module, probably better to use a switch expression and no `default` to ensure compile time exhaustive checking.

Reply:
@haidubogdan Please add  all cases instead of `default` because you changed to `javac.release=17`.